src/ServiceControl.MultiInstance.AcceptanceTests/SagaAudit/When_sending_saga_audit_to_main_instance.cs

```diff
@@ -66,6 +83,57 @@ public async Task Saga_history_can_be_fetched_from_main_instance()
             Assert.AreEqual("Send", sagaStateChange.InitiatingMessage.Intent);
         }
 
+        [Test]
+        public async Task
+        Check_fails_if_no_audit_retention_is_configured()
+        {
+            //Ensure custom checks are enabled
+            CustomServiceControlSettings = settings =>
+            {
+                settings.DisableHealthChecks = false;
+            };
+
+            //Override the configuration of the check in the container in order to make it run more frequently for testing purposes.
+            CustomEndpointConfiguration = config =>
+            {
+                config.RegisterComponents(registration =>
+                {
+                    registration.ConfigureComponent((builder) => new AuditRetentionCustomCheck(builder.Build<IDocumentStore>(), builder.Build<Settings>(), TimeSpan.FromSeconds(10)), DependencyLifecycle.SingleInstance);
+                });
+            };
+
+            SingleResult<EventLogItem> customCheckEventEntry = default;
+
+            SingleResult<SagaHistory> sagaAudiDataInMainInstanceIsAvailableForQuery = default;
+
+            await Define<MyContext>()
+                .WithEndpoint<SagaEndpoint>(b => b.When((bus, c) => bus.SendLocal(new MessageInitiatingSaga { Id = "Id" })))
+                .Done(async c =>
+                {
+                    if (!c.SagaId.HasValue)
+                    {
+                        return false;
+                    }
+
+                    if (sagaAudiDataInMainInstanceIsAvailableForQuery?.HasResult == false)
```

Comment:
The left-hand side of this expression will be `null`, so the condition will never be true. The test still works because the event log item below still gets generated and that's all we assert on once the scenario finishes. You could get rid of this block altogether or turn it into `!= true`.

Reply:
My bad. I wanted to make it a bit faster by not querying /api/eventlogitems/ every time but didn't get the logic right. Just pushed a fix. It saves 2 seconds.